### PR TITLE
Add callback context in cellular HAL callback APIs

### PR DESCRIPTION
**Pull request by chinglee-iot**
Description
-----------

* Add callback context in cellular HAL callback APIs
* Fix cellular manager compile warning

Checklist:
----------

- [ ] I have tested my changes. No regression in existing tests.
- [ ] My code is Linted.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: libraries/abstractions/cellular/test/test_cellular_api.c [hunk 1]
```diff
@@ -283,7 +286,8 @@ static void prvGenericCallback( const char * pRawData )
 
 /* PDN event callback function. */
 static void prvPdnEventCallback( CellularUrcEvent_t urcEvent,
-                                 uint8_t contextId )
+                                 uint8_t contextId,
```
Comment (DanielYEHsieh) [inline libraries/abstractions/cellular/test/test_cellular_api.c:289]: should we add const here?

File: libraries/abstractions/cellular/test/test_cellular_api.c [hunk 2]
```diff
@@ -354,7 +358,8 @@ static void prvCellularSocketOpenCallback( CellularUrcEvent_t urcEvent,
 /*-----------------------------------------------------------*/
 
 /* Modem event callback function. */
-static void prvCellularModemEventCallback( CellularModemEvent_t modemEvent )
+static void prvCellularModemEventCallback( CellularModemEvent_t modemEvent,
```
Comment (DanielYEHsieh) [inline libraries/abstractions/cellular/test/test_cellular_api.c:361]: should we add const here?

File: libraries/abstractions/cellular/test/test_cellular_api.c (hunk 1 above)
Comment (chinglee-iot) [reply libraries/abstractions/cellular/test/test_cellular_api.c:289]: This is not modified in the commit. I suggest we have another commit review over the test_cellular_api.c.

**Comment by DanielYEHsieh** [comment]
I've noticed a few APIs prototype seems required another round modification which is out of scope of the purpose of this PR.
In this PR prototypes is ignored

**Review by DanielYEHsieh — APPROVED** [review]

**Review by andysun2015 — APPROVED** [review]

**Pull request merged**